src/main/scala/com/intersys/mdc/challenge/exercises/problems/Problem4.scala

```diff
@@ -55,15 +59,42 @@ case object Problem4 extends Problem {
 
           // A) Implicit class here with:
           // def asInt: Option[Int] = ???
-          ???
 
           // B) Implement the calculate method.
           // def calculate(ops: String, a: Int, b: Int): Option[Int] = ???
-          def calculate(ops: String, a: Int, b: Int): Option[Int] = ???
+          def calculate(ops: String, a: Int, b: Int): Option[Int] = {
+            ops match {
+              case "sum" => Some(a+b)
+              case "subtraction" => Some(a-b)
+              case "multiplication" => Some(a*b)
+              case "division" => try {
+                Some(a/b)
+              } catch {
+                case _: Exception => None
+              }
+              case _ => None
+            }
+          }
 
           // C) Complete the challenge response variable.
           // val challengeResponse: Option[Calculation] = ???
-          val challengeResponse: Option[Calculation] = ???
+          val challengeResponse: Option[Calculation] = {
+            val a = params("a").asInt
+            val b = params("b").asInt
+            val operation = params("operation")
+
+            if(a == None || b == None) {
```

Comment:
👎 Could have used for comprehensions (flatMap and map) for the Option monad. Not idiomatic scala code.